dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/WorkflowItemRestRepository.java

```diff
@@ -0,0 +1,335 @@
+            //the value in the position 0 is a null value
+            String[] path = op.getPath().substring(1).split("/", 3);
+            if (OPERATION_PATH_SECTIONS.equals(path[0])) {
+                String section = path[1];
+                evaluatePatch(context, request, source, wsi, section, op);
+            } else {
+                throw new PatchBadRequestException(
+                    "Patch path operation need to starts with '" + OPERATION_PATH_SECTIONS + "'");
+            }
+        }
+        wis.update(context, source);
+    }
+
+    private void evaluatePatch(Context context, HttpServletRequest request, XmlWorkflowItem source,
+                               WorkflowItemRest wsi, String section, Operation op) {
+        SubmissionConfig submissionConfig =
+            submissionConfigReader.getSubmissionConfigByName(wsi.getSubmissionDefinition().getName());
+        for (int stepNum = 0; stepNum < submissionConfig.getNumberOfSteps(); stepNum++) {
+
+            SubmissionStepConfig stepConfig = submissionConfig.getStep(stepNum);
+
+            if (section.equals(stepConfig.getId())) {
+                /*
+                 * First, load the step processing class (using the current
+                 * class loader)
+                 */
+                ClassLoader loader = this.getClass().getClassLoader();
+                Class stepClass;
+                try {
+                    stepClass = loader.loadClass(stepConfig.getProcessingClassName());
+
+                    Object stepInstance = stepClass.newInstance();
+
+                    if (stepInstance instanceof AbstractRestProcessingStep) {
+                        // load the JSPStep interface for this step
+                        AbstractRestProcessingStep stepProcessing =
+                            (AbstractRestProcessingStep) stepClass.newInstance();
+                        stepProcessing.doPreProcessing(context, source);
+                        stepProcessing.doPatchProcessing(context, getRequestService().getCurrentRequest(), source, op);
+                        stepProcessing.doPostProcessing(context, source);
+                    } else {
+                        throw new PatchBadRequestException(
+                            "The submission step class specified by '" + stepConfig.getProcessingClassName() +
+                            "' does not extend the class org.dspace.submit.AbstractProcessingStep!" +
+                            " Therefore it cannot be used by the Configurable Submission as the <processing-class>!");
+                    }
+
+                } catch (Exception e) {
+                    log.error(e.getMessage(), e);
+                }
+            }
+        }
+    }
+
+    @Override
+    protected void delete(Context context, Integer id) {
+        XmlWorkflowItem witem = null;
+        try {
+            witem = wis.find(context, id);
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
```

Comment:
Again, a more descriptive error message please.  Something like `"Could not locate workspace item with id" + id`

Reply:
resolving as agreed